File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportJava/Persistence/src/main/java/transport/persistence/repository/IRepository.java
package transport.persistence.repository;

import transport.model.Entity;

import java.util.List;

public interface IRepository<ID, E extends Entity<ID>> {
    ID add(E entity);
    void delete(ID id);
    void update(E entity);
    E findById(ID id);
    List<E> findAll();
}

File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportC#/Persistence/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FirmaTransportC_.Model;
namespace FirmaTransportC_.Persistence
{
    public interface IRepository<ID, E> where E : Entity<ID>
    {
        ID Add(E entity);
        void Delete(ID id);
        void Update(E entity);
        E FindOne(ID id);
        List<E> FindAll();
    }
}
